### DISCUSSION TITLE
Keycloak user events saved are not clearing past the expiry time set

### DISCUSSION THREAD

**Discussion by pravsjv** [keycloak/keycloak#14825] [Q&A] [answered]
Hello everyone,

I'm saving the user and admin events in my keycloak setup and set an expiry time(for testing, I set 1min). I managed to save some events and left it as it is for few days and checked if they are cleared, but I see them exist in Event_Entity table.
Also when there are very high volume of events saved, and try to click on "clear events", its not clearing all the events.

Are these expected behaviors?

**Comment by sschu** [comment]
Which version were you using? There was recently a bug that made event expiration not work. It is fixed in current master and 19.0.3.

**Reply by pravsjv** [reply]
I tested it in 19.0.1 recently and as well we face this issue in older version of keycloak in production. If its fixed in 19.0.3 it makes sense as I've tested it in 19.0.1

**Reply by pravsjv** [reply]
Can I have the code commit link if possible to check if I can backport the changes?

**Reply by sschu** [reply]
Is is this one: https://github.com/keycloak/keycloak/commit/97b999614dbb6ef8910c01370fe12b6620316fe4

However, I think that bug was introduced with 19.0. With 18.0.2, everything definitely works. So maybe your issue is a different one.

**Comment by pravsjv** [comment, marked as answer]
This one is working fine without any issues in the latest releases.

**Comment by mabhisheksingh** [comment]
HI @pravsjv 

I have designed a new approach, explained below. Can you test and let me any changes required 
In the meantime, you can use an environment-based configuration system similar to what’s implemented in this repo:
https://github.com/mabhisheksingh/keycloak-webhook-bridge

Our design allows you to specify the target system’s URL and IP via environment variables. The service will then send requests to the specified endpoint, including a marker to indicate whether the event is an admin or user event.

Feel free to let me know if you need more details or examples of the ENV setup!
